### PR TITLE
Feature/#59 Student Module: Training TOC

### LINKED ISSUE #59: [Student Module] Training Content viewer
On the student module we want implement the following page:

Page View

This is the main topic page, we will start by showing the main table of content page (later on in a separate case we will add internal navigation and stuff like that).

Steps to implement this:

- Under _./src/rest-api_ create a subfolder called _student_ (we will just create separate rest-api per module.

- Then under that path create a _training.ts_ this will include a method called _getTrainingMainToc_:

```javascript
public getTrainingMainToc(int trainingId) : Promise<string>
{
   const testMarkdown = `
# Module 1


**Layout and css**

Some test description


# Heading 1

**Work in progress: we are working on closing a first release of this library, we expect to register it in npm in the following two three weeks, in the mean time you can download the source code and run the samples following the instructions below.**
   `;

return Promise.Resolve(testMarkdown );
}


```

- Now in the reducers side let's create on _studentTraining_ where we will store the current markdown (one thing here to discuss with @nasdan, I think we should create subfolders in the reducers side per module).

- Let's create the action to fetch the markdown from the rest api (async action, thunk), this action will take as input parameter the training ID (following TDD).

- On the component side, let's create containers and components (following TDD), we have to add
as well glue: navigation, once you make the login you navigate to a training selection (that's where the ID comes, but right now we will ignore this), then we will jump into view training main page (our page), let's start by hardcoding the ID in the component.

Since this a medium complexity task I will assign @Nasdan to provide support on it.

### PR DESCRIPTION
**Pull request by crsanti**
#### Summary 
- Added page, container, actions, reducers, routing constants and navigation.
- Added [marksy](//github.com/cerebral/marksy) for Markdown to React Components ([markdown-to-react-components](//github.com/christianalfoni/markdown-to-react-components) is deprecated).

This closes #59.

### CODE REVIEW

File: src/rest-api/student/trainingTocMockData.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { TrainingTOC } from '../../model/student/trainingToc';
+
+export const trainingTOCMockData: TrainingTOC[] = [
```
Comment (brauliodiez) [inline src/rest-api/student/trainingTocMockData.ts]: Let's add here the real training description we created this morning

File: tslint.json [hunk 2]
```diff
@@ -71,6 +71,10 @@
         "no-internal-module": true,
         "no-trailing-whitespace": true,
         "no-unsafe-finally": true,
+        "no-unused-variable": [
+          true,
+          "react"
+        ],
```
Comment (brauliodiez) [inline tslint.json]: What is this setting for?

**Comment by brauliodiez** [comment]
Add a temp link to navigate from training list to edit TOC

**Comment by brauliodiez** [comment]
Adding mock data (already pushed), I have realized there is an issue with spanish tilde, e.g. Máster, do we need to escape the tildes? is an issue with marksy? Let's open an issue on our project for further checking.

**Comment by brauliodiez** [comment]
The sample text I have uploaded generate a linter warning 

````
 Exceeds maximum line length of 120
```

Can we disable this check for this file?

File: tslint.json (hunk 2 above)
Comment (crsanti) [inline tslint.json]: To detect unused created or imported vars (except React when tslint detects any JSX element).

**Comment by crsanti** [comment]
@brauliodiez The issue with encoding was because of the copy & paste of the source encoding. I've fixed it.

**Comment by brauliodiez** [comment]
give me a buzz when this is ready for a second review

**Comment by crsanti** [comment]
It's ready for review :)

**Comment by brauliodiez** [comment]
Warnings removed? still got them gong to remove node_modules and test again

**Pull request merged**